system/Autoloader/FileLocator.php

```diff
@@ -136,15 +136,18 @@ public function locateFile(string $file, string $folder = null, string $ext = 'p
 		// Check each path in the namespace
 		foreach ($paths as $path)
 		{
+			// Ensure trailing slash
+			$path = rtrim($path, '/') . '/';
+			
 			// If we have a folder name, then the calling function
 			// expects this file to be within that folder, like 'Views',
 			// or 'libraries'.
 			if (! empty($folder) && strpos($path . $filename, '/' . $folder . '/') === false)
 			{
-				$path .= $folder;
+				$path .= trim($folder, '/') . '/';
```

Comment:
Should this be an `rtrim`? We only care about the trailing slash, right?

Reply:
Conceivably `$folder` could have a preceding slash. We could let that be a user input error and go with `rtrim()`, or stick with this. At this point `$path` is guaranteed a trailing slash.

Reply:
Hmm... will have to look later and see where all it's being used in the system, but I can imagine possibilities where someone wants to reference a full unix path, which would have a preceding slash that would need to stay.

Reply:
`$folder` is by definition a path within an existing namespace so will always be appended to `$path` (which has the trailing slash) so I think the only thing a preceding slash would get you is doubled slashes, which I'm pretty sure *nix systems treat as a single slash anyways.

Reply:
Also per your comment on the original issue (https://github.com/codeigniter4/CodeIgniter4/issues/1866) `locateFile` is only used internal to FileLocator, so should be fine.

Reply:
Ah, thanks for doing the research for me! Wasn't going to have time until tonight. So I'm cool with this then :)